lib/model/content.dart

```diff
@@ -576,27 +574,69 @@ class ImagePreviewNode extends BlockContentNode {
   bool operator ==(Object other) {
     return other is ImagePreviewNode
       && other.srcUrl == srcUrl
-      && other.thumbnailUrl == thumbnailUrl
+      && other.thumbnail == thumbnail
       && other.loading == loading
       && other.originalWidth == originalWidth
       && other.originalHeight == originalHeight;
   }
 
   @override
   int get hashCode => Object.hash('ImagePreviewNode',
-    srcUrl, thumbnailUrl, loading, originalWidth, originalHeight);
+    srcUrl, thumbnail, loading, originalWidth, originalHeight);
 
   @override
   void debugFillProperties(DiagnosticPropertiesBuilder properties) {
     super.debugFillProperties(properties);
     properties.add(StringProperty('srcUrl', srcUrl));
-    properties.add(StringProperty('thumbnailUrl', thumbnailUrl));
+    properties.add(DiagnosticsProperty<ImageThumbnailLocator>('thumbnail', thumbnail));
     properties.add(FlagProperty('loading', value: loading, ifTrue: "is loading"));
     properties.add(DoubleProperty('originalWidth', originalWidth));
     properties.add(DoubleProperty('originalHeight', originalHeight));
   }
 }
 
+/// Data to locate an image thumbnail,
+/// and whether the image has an animated version.
+///
+/// Use [ImageThumbnailLocatorExtension.resolve] to obtain a suitable URL
+/// for the current UI need.
+@immutable
+class ImageThumbnailLocator extends DiagnosticableTree {
+  ImageThumbnailLocator({
+    required this.defaultFormatSrc,
+    required this.animated,
+  }) : assert(!defaultFormatSrc.isAbsolute),
```

Comment:
This has a surprising semantics. From the implementation:
```dart
  bool get isAbsolute => hasScheme && !hasFragment;
```

Presumably that matches what some old RFC says "absolute" means. But if we expand this out to `assert(!defaultFormatSrc.hasScheme || defaultFormatSrc.hasFragment)`, that's clearly not what we want here!

Reply:
Wacky!